aiokafka/consumer/assignors.py

```diff
@@ -0,0 +1,24 @@
+from aiokafka.coordinator.assignors.abstract import AbstractPartitionAssignor
+import abc
+
+
+class AbstractStaticPartitionAssignor(AbstractPartitionAssignor):
+    """
+    Abstract assignor implementation that also supports static assignments (KIP-345)
+    """
+
+    @abc.abstractmethod
+    def assign(self, cluster, members,  # lgtm[py/inheritance/signature-mismatch]
+               member_group_instance_ids):
```

Comment:
Based on the [KIP-345 proposal](https://cwiki.apache.org/confluence/display/KAFKA/KIP-345%3A+Introduce+static+membership+protocol+to+reduce+consumer+rebalances), I don't think this class is actually necessary, since the client side need only provide the `group.instance.id` with the appropriate `Join/Sync request/responses` to utilize static membership.

With that said, I'm removing this class. I tested locally and the tests still pass.

Reply:
Small update, I looked at the original commits over at the Robinhood fork, and it looks like this class was added as an interface for containerized deployments, so that others could extend the class and use the `group.instance.id` mapping in their implementation. See the [original commit message here, specifically paragraph three](https://github.com/robinhood/aiokafka/commit/cd44bdcf57b4aa6f675783ed1043b2d643b0ac54).

With that said, it may be worth keeping so anyone that is using an older fork to utilize KIP-345 could move over to this version of `aiokafka` with ease (in the case that they implement their own StaticPartitionAssignor.)

@ods What do you think? Any objections to keeping this?

Reply:
I'd prefer to keep it if it's useful. But we'll have to solve the problem with interface mismatch. I think we could move it to separate PR to not block the rest.

Reply:
Agreed, this can be re-added in another PR to not block the rest of this work. I'll leave the PR as-is for now, since I already removed the code block from this PR.